apps/src/storage/dataBrowser/dataVisualizer/VisualizerModal.jsx

```diff
@@ -196,9 +198,39 @@ class VisualizerModal extends React.Component {
     return options.join(', ');
   }
 
+  countCells = memoize((records, tableColumns, chartType) => {
+    if (chartType !== ChartType.CROSS_TAB) {
+      return undefined;
+    } else {
+      return records.reduce((bucket, record) => {
+        tableColumns.forEach(column => {
+          bucket[column] ||= new Set();
+          bucket[column].add(record[column]);
+        });
+        return bucket;
+      }, new Set());
+    }
+  });
+
+  filterTableColumnsByCount = memoize(
+    (tableColumns, countedCells, selectedColumn1) => {
+      if (!countedCells) {
+        return tableColumns;
+      } else {
+        const xCount = selectedColumn1 ? countedCells[selectedColumn1].size : 0;
+        return tableColumns.filter(
+          column =>
+            countedCells[column].size < MAX_CROSSTAB_COLUMNS &&
+            xCount * countedCells[column].size < MAX_CROSSTAB_CELLS
+        );
+      }
+    }
+  );
+
```

Comment:
And this one will filter out our table columns based upon whether they exceed our thresholds.

If there are no countedCells, we filter nothing. If there is a `selectedColumn1`, then we also filter on the cross product to count the number of cells.